LinGUI: More detailed notifications

robxnano opened the pull request:
**Description of Change:**

Adds new option to show notification as each individual job is completed as well as all jobs are done. Also shows the number of completed (and failed) items when the queue is finished, and shows notifications when encoding is paused automatically. Finally there are some minor bug fixes. This also helps address #663.

The notification text is subject to change if you feel it needs to be reworded.

**Tested on:**

- [x] Windows 10+  (via MinGW)
- [ ] macOS 10.13+
- [x] Ubuntu Linux

**Screenshots**
Notifications:
![Screenshot from 2023-11-25 17-45-01](https://github.com/HandBrake/HandBrake/assets/89391914/c1b98327-3e61-4a1a-9bac-0eecae7086b9)
![Screenshot from 2023-11-25 17-45-12](https://github.com/HandBrake/HandBrake/assets/89391914/0e79c67a-196e-4453-80a0-abd378f334fa)
![Screenshot from 2023-11-25 17-45-41](https://github.com/HandBrake/HandBrake/assets/89391914/4f6a0868-4385-4c68-a890-deb314f157a4)
![Screenshot from 2023-11-25 17-45-58](https://github.com/HandBrake/HandBrake/assets/89391914/794ea4d2-3fb5-49bf-9bc1-dbd3e04c4db9)

Notification Settings:
![Screenshot from 2023-11-24 23-51-50](https://github.com/HandBrake/HandBrake/assets/89391914/e706b1f1-1316-466f-a520-cadedde75f8b)

robxnano committed:
LinGUI: Improve notifications

- Notify when encoding is paused automatically
- Add option to notify when each encode completes
- Show number of encodes on queue finished notification
- Rearrange Settings page to accommodate new options

robxnano committed:
LinGUI: Add separate preference for notifications

robxnano committed:
LinGUI: Inhibit logout when encoding

When an encode is in progress, attempting to log out or shut down
the computer will show a warning to remind the user that their
encoding task is not yet finished. Only on certain DEs.

galad87 commented:
Looks good to me, I'll leave the decision about the wording and the infamous cocktail phrase to someone else 😅

Can the first three commits be back ported to the 1.7.x branch?

bradleysepos commented:
We should definitely keep the phrase, but I've always been a proponent of making it configurable.

robxnano committed:
LinGUI: Add option for custom notification message

This option allows users to change the title of the queue complete
notification.

galad87 commented:
Thank you!
